app/javascript/components/supplemental_application/actions.js

```diff
@@ -6,14 +6,20 @@ import { isEmpty, find, isEqual, every } from 'lodash'
 export const updateApplication = async (application, prevApplication) => {
   const primaryApplicantContact = application.applicant && application.applicant.id
   const promises = [updateLease(application['lease'], primaryApplicantContact, application['id'])]
-
   // Concat lease promise with rental assistances
   promises.concat(updateUnsavedRentalAssistances(application, prevApplication))
   const applicationApi = domainToApi.buildApplicationShape(application)
-  promises.push(apiService.submitApplication(applicationApi))
+  promises.push(apiService.submitApplication(applicationApi, true))
   const responses = await Promise.all(promises)
 
-  return every(responses, (promise) => promise !== false)
+  if (every(responses, (promise) => promise !== false)) {
+    const [ { lease }, { application } ] = responses
+    if (application) {
+      application.lease = lease
```

Comment:
The responses come back as 200's no matter what, but they return either a value or false, so we check to see the responses are not just false like before and then return the application with information otherwise false